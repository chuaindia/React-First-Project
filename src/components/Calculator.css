* {
  padding: 0;
  margin: 0;
}

.mainframe {
  background: white;
  display: flex;
  flex-direction: column;
  align-items: center;
}

.mainframe h2 {
  text-align: left;
}

.screen {
  width: 93.1%;
  height: 100%;
  border: 2px solid black;
  color: white;
  padding: 10px;
  text-align: right;
  background: gray;
}

.row {
  display: grid;
  grid-template-columns: repeat(4, 1fr);
}

button {
  border: 1px solid black;
  padding: 10px 20px;
  color: black;
  text-align: center;
  text-decoration: none;
  font-size: 2rem;
}

button:hover {
  color: aqua;
  background-color: black;
}

button:active {
  background-color: grey;
}

.buttoncolor {
  background: orangered;
}

.zero {
  grid-column: 1/ span 2;
}
