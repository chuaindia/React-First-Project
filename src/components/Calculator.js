import React from 'react';

const buttons = ['AC', '+/-', '%', 'รท', '7',
  '8', '9', 'x', '4', '5', '6', '-',
  '1', '2', '3', '+',
  '0', '.', '='];

const MakeArray = () => {
  const arrayForButtons = [];
  buttons.forEach((button) => arrayForButtons.push(
    <button className="calc-btns" type="button">{button}</button>,
  ));
  return arrayForButtons;
};

class Calculator extends React.Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    return (
      <>
        <div id="calculator">
          <div id="display-calculator" />
          <input />
        </div>
        <div id="calculator-buttons">
          <MakeArray />
        </div>
      </>
    );
  }
}

export default Calculator;
