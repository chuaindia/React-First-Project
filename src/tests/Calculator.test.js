import React from 'react';
import renderer from 'react-test-renderer';
import Calculator from '../components/Calculator';

it('iIt will render the buttons', () => {
  const screen = renderer
    .create(<Calculator />)
    .toJSON();
  expect(screen).toMatchSnapshot();
});
