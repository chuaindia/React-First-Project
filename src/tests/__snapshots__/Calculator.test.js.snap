// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`iIt will render the buttons 1`] = `
<div
  className="main-frame"
>
  <h1>
    Math Magician
  </h1>
  <h2>
     Let us do some basic Mathematics Operations
  </h2>
  <div
    className="main"
  >
    <div
      className="calculator"
    >
      <div
        className="screen"
      >
        0
      </div>
      <div
        className="first row"
      >
        <button
          onClick={[Function]}
          type="button"
        >
          AC
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          +/-
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          %
        </button>
        <button
          className="buttoncolor"
          onClick={[Function]}
          type="button"
        >
          รท
        </button>
      </div>
      <div
        className="second row"
      >
        <button
          onClick={[Function]}
          type="button"
        >
          7
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          8
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          9
        </button>
        <button
          className="buttoncolor"
          onClick={[Function]}
          type="button"
        >
          x
        </button>
      </div>
      <div
        className="third row"
      >
        <button
          onClick={[Function]}
          type="button"
        >
          4
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          5
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          6
        </button>
        <button
          className="buttoncolor"
          onClick={[Function]}
          type="button"
        >
          -
        </button>
      </div>
      <div
        className="fourth row"
      >
        <button
          onClick={[Function]}
          type="button"
        >
          1
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          2
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          3
        </button>
        <button
          className="buttoncolor"
          onClick={[Function]}
          type="button"
        >
          +
        </button>
      </div>
      <div
        className="fifth row"
      >
        <button
          className="zero"
          onClick={[Function]}
          type="button"
        >
          0
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          .
        </button>
        <button
          className="buttoncolor"
          onClick={[Function]}
          type="button"
        >
          =
        </button>
      </div>
    </div>
  </div>
</div>
`;
